### PR TITLE
Add multiselection parameter to dropdown options

### PR DESCRIPTION
**Pull request by ialarmedalien**
Adding a "multiselection" parameter to allow dropdowns to have multi select enabled. The dynamic dropdown type already has a multi selection param so I used the same format for the dropdown type.

Note that full implementation requires modification of the narrative to implement the new spec; this will be done in a separate (obviously!) PR on the narrative repo.

### CODE REVIEW

File: src/us/kbase/narrativemethodstore/db/test/NarrativeMethodDataTest.java [hunk 1]
```diff
@@ -104,56 +105,71 @@ public void testGroupOneOk() throws Exception {
...
+        // 6 has it enabled
+
+        final DropdownOptions ddoSingle = nmd.getMethodSpec().getParameters().get(3)
+                .getDropdownOptions();
+        assertThat("incorrect multiselect", ddoSingle.getMultiselection(), is(0L));
+
+        final DropdownOptions ddoMulti = nmd.getMethodSpec().getParameters().get(6)
+                .getDropdownOptions();
+        assertThat("incorrect multiselect", ddoMulti.getMultiselection(), is(1L));
+    }
```
Comment (ialarmedalien) [inline src/us/kbase/narrativemethodstore/db/test/NarrativeMethodDataTest.java:121]: git is dumb and doesn't realise that I inserted this code block and didn't change the next two tests.

**Review by briehl — CHANGES_REQUESTED** [review]
Mostly, I think some updated docs are needed. Otherwise, it looks great to me.

File: NarrativeMethodStore.spec [hunk 2]
```diff
@@ -184,6 +184,7 @@ module NarrativeMethodStore {
 
     typedef structure {
         list<DropdownOption> options;
+        boolean multiselection;
```
[the remark this answers is not in the document]
Comment (ialarmedalien) [reply NarrativeMethodStore.spec:187]: OK, done!

[the remark this answers is not in the document]
Comment (briehl) [reply NarrativeMethodStore.spec:204]: Great!

**Review by briehl — APPROVED** [review]

**Pull request merged**